cpp/src/gandiva/decimal_type_util.cc

```diff
@@ -59,7 +62,8 @@ Status DecimalTypeUtil::GetResultType(Op op, const Decimal128TypeVector& in_type
       break;
 
     case kOpDivide:
-      result_scale = std::max(kMinAdjustedScale, s1 + p2 + 1);
+      result_scale = use_redshift_rules ? std::max(4, s1 + p2 - s2 + 1)
```

Comment:
Can we define a named constant instead of using magic number `4` directly?

Reply:
done